widget/diagramwidget/diagram.go

```diff
@@ -0,0 +1,251 @@
+package diagramwidget
+
+import (
+	"image/color"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/driver/desktop"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+var Globaldiagram *DiagramWidget
```

Comment:
I appreciate that this is part of a workaround as described below but we should not have public global APIs.
The workaround should not use globals anyhow as the refresh should apply to the window of the widget - you may have more than one instance in an application.

Reply:
Made local